html/captive-portal/lib/captiveportal/PacketFence/Form/Authentication.pm

```diff
@@ -107,6 +107,35 @@ sub check_aup {
     return ;
 }
 
+=head2 check_telephone_form
+
+Check telephone form
+
+=cut
+
+sub check_telephone_form {
+    my ($self, $field) = @_;
+    if($self->app->request->method eq "POST"){
+        if ( $field->value !~ /^([0-9]+[0-9-\.]+[0-9]+)$/ ) {
+            $field->add_error("Enter a valid telephone <em>number</em>");
+            $self->app->flash->{error} = "Telephone number is not valid";
+        }
+    }
+}
+
+=head2 check_telephone
+
+Check that the telephone is valid
+
+=cut
+
+sub check_telephone {
```

Comment:
Couldn't you get rid of that one and just call check_telephone_form directly

Reply:
I followed existing convention and I'm not sure if it's needed. The intermediate method may be in place so they can be overloaded.

Reply:
@julsemaan shall I leave it as-is, or remove the intermediate method?

Reply:
Lets leave it as is so it can be overloaded